flutter/devtools
# Convert ansi_up.js to dart (https://github.com/flutter/devtools/issues/1181)

### Summary
This change converts most of the `ansi_up.js` file to dart so that we will be able to use it for the logging and debugging pages in flutter web as well.

Updates include:
- Moving JS functions into `ansi_up.dart`
- Adding unit test for main ansi function
- Removing references to JS file and removing the JS file

### Testing
To run the unit test:
```
cd packages/devtools_app
flutter test test/ansi_up_test.dart
```

To check that dart:html continues to work:
- Follow instructions in CONTRIBUTING.md to run dart:html web
- Run an app with logging messages including ansi codes, or run a modified version of the fixture file `color_console_output_app` which does this (as in https://github.com/helin24/devtools/commit/69dd41949be5665e33a78bd88442b16052a350d1)
- Verify that color codes appear in logging.
